src/postpredict/dependence.py

```diff
@@ -0,0 +1,83 @@
+import abc
+
+import polars as pl
+
+from .util import argsort_random_tiebreak
+
+
+class TimeDependencePostprocessor(abc.ABC):
+    def __init__(self):
+        pass
+
+
+    @abc.abstractmethod
+    def fit(self, **kwargs):
+        """
+        Fit a model for temporal dependence across prediction horizons
+
+        Returns
+        -------
+        None
+        """
+
+
+    @abc.abstractmethod
+    def transform(self, model_out, **kwargs):
+        """
+        Apply an estimated time dependence model to sample predictions to induce
+        dependence across time in the predictive samples.
+        
+        Parameters
+        ----------
+        model_out: polars dataframe with sample predictions that do not
+        necessarily capture temporal dependence.
+        
+        Returns
+        -------
+        A copy of the model_out parameter, with sample indices updated so that
+        they reflect the estimated temporal dependence structure.
+        """
+    
+    
+    def apply_shuffle(self, wide_model_out, value_cols, templates):
```

Comment:
To get the full benefit of running mypy type checking, you might consider adding types to variables used throughout the code base (some people add types to _every_ variable, but I've found that just typing function parameters strikes a good balance).

So this line could read: 
```python
def apply_shuffle(self, wide_model_out: pl.DataFrame, value_cols: list[str], templates: np.array) -> pl.DataFrame:
```

(Not sure if `templates` is actually a numpy array because I couldn't infer that from the docstring. That's another way explicitly typing is helpful imo: it allows people unfamiliar with the code base to understand it faster)

Mypy will alert you if your code doesn't align with the type assigned to the variable. For example, if you type `wide_model_out` as pl.DataFrame, you'll get warnings if you use that variable incorrectly:

<img width="520" alt="image" src="https://github.com/user-attachments/assets/6b04bba3-93cb-43e6-a801-49012a81b4ee">